lib/ex_doc/formatter/html/templates.ex

```diff
@@ -246,11 +246,30 @@ defmodule ExDoc.Formatter.HTML.Templates do
     |> elem(0)
   end
 
+  @class_regex ~r/class="([^"]+)/
+  @class_separator " "
   defp link_heading(match, _tag, _title, "", _prefix), do: match
 
-  defp link_heading(_match, tag, title, id, prefix) do
+  defp link_heading(match, tag, title, id, prefix) do
```

Comment:
REVIEWER: This change is mainly to preserve the previous `class` attributes from the headers, in case there is one. Now with the _admonition_ text block, we could inject classes. The supported classes so far are: `warning`, `info`, `error`.

Reply:
This is a great explanation. Can you please add it as code comments? :)